### ISSUE TITLE
Mix xref callers does not include calls within the module that defines the function

### ISSUE THREAD

**Issue by balduncle** [elixir-lang/elixir#9198]
### Environment

* Erlang/OTP 22 [erts-10.4.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]
* Elixir 1.9.0 (compiled with Erlang/OTP 22)
* macOS Mojave

### Current behavior

When a module defines a function and also calls the function, `mix xref callers Module.function` does not return the call.  For example...

```
defmodule Xref do
  def hello do
    "hello #{world()}"
  end

  def world do
    "world"
  end
end
```

```
$ mix xref callers Xref.world
$ 
```

This can also be observed in iex.

```
iex> Mix.Tasks.Xref.calls() |> Enum.filter(fn %{callee: {m, _, _}} -> m == Xref end)
[]
```

### Expected behavior

I would expect calls within the same file as the function definition to be returned.  The doc string for `Mix.Tasks.Xref.calls` says `Returns a list of information of all the function calls in the project.`.  This seems to imply that it would include calls within the same module that defined the function.

**Comment by balduncle** [comment]
In case it is helpful background information, I ran into this problem while working on adding rename support to ElixirLS in https://github.com/JakeBecker/elixir-ls/issues/22.  ElixirLS relies on `Mix.Tasks.Xref` to find all references to a function.

**Comment by josevalim** [comment]
`xref` is about cross references, that's why it doesn't track local information, but our hope is to have a streamlined mechanism for consuming compilation metadata in Elixir v1.10 and we can include local calls in there.

**Comment by balduncle** [comment]
Thanks @josevalim.  Is there a good way to get plugged into conversations about IDE/tooling related features being added to Elixir?  As I've been thinking about how to handle various edge cases in ElixirLS I keep thinking that I want to make sure they are being discussed before features are added to Elixir.

**Comment by josevalim** [comment]
elixir-lang-core is where those discussions happen. :) But i will also post a follow up here once the PR is in place.

**Comment by balduncle** [comment]
Brilliant.  🤦‍♂  I have joined the mailing list.  Thank you.

**Comment by josevalim** [comment]
Closing in favor of #9247.
